src/TorchSharp/NN/Module.cs

```diff
@@ -256,16 +258,14 @@ private void _toEpilog(ScalarType? dtype, Device device)
                                 .DetachFromDisposeScope() as Parameter;
 
                             // Copy the gradient over as well, if it exists
-                            var grad = param.grad();
                             if (grad is not null) {
-                                p.set_grad(grad.to(paramType, device ?? param.device)
-                                                .with_requires_grad(grad.requires_grad)
-                                                .MoveToOtherDisposeScope(p));
+                                p.grad = grad.to(paramType, device ?? param.device)
+                                    .with_requires_grad(grad.requires_grad)
+                                    .MoveToOtherDisposeScope(p);
                             }
 
-                            // Dispose the param and gradient
+                            // Dispose the param
                             param.Dispose();
-                            grad?.Dispose();
```

Comment:
Does disposing the param automatically dispose the grad?

Reply:
It's [line 242](https://github.com/dotnet/TorchSharp/pull/1299/files/5ac8a73e5f5247791dcdc3e6cae1d9cbf9bcf9eb#diff-c9fe99719d1977951a66f323175f5bcd3ba449f30a3711bbc84defa26e924460R242) disposing that.

Reply:
It's [line 242](https://github.com/dotnet/TorchSharp/pull/1299/files/5ac8a73e5f5247791dcdc3e6cae1d9cbf9bcf9eb#diff-c9fe99719d1977951a66f323175f5bcd3ba449f30a3711bbc84defa26e924460R242) disposing that.